api/src/main/java/org/openmrs/module/referencedemodata/obs/DemoObsGenerator.java

```diff
@@ -73,22 +86,39 @@ public DemoObsGenerator(DemoDataConceptCache conceptCache) {
 	
 	public void createDemoVitalsObs(Patient patient, Encounter encounter, Location location) {
 		for (NumericObsValueDescriptor vitalsDescriptor : getVitalsDescriptors()) {
-			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, location);
+			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, encounter.getEncounterDatetime(), location);
 		}
 	}
 	
 	public void createDemoLabObs(Patient patient, Encounter encounter, Location location) {
-		for (NumericObsValueDescriptor labDescriptor : getLabDescriptors()) {
-			createNumericObsFromDescriptor(labDescriptor, patient, encounter, location);
+		List<Concept> labConcepts = new ArrayList<>();
+		// Add panels thrice to increase chance of being picked
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("Test"));
+		
+		Concept c = (Concept) randomArrayEntry(labConcepts.toArray());
+		if (c.getConceptClass() != null && "LabSet".equalsIgnoreCase(c.getConceptClass().getName()) && c.getSet()) {
```

Comment:
This condition means we'll do weird things if we end up with a concept with the class "LabSet" that is not a concept set, no?

Reply:
This should not happen and in any case it does, there is catch up the stack that takes care of this, in the `Obs createDemoLabObs(Patient, Encounter, Date, Location, Concept)` method.